flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/table/options/ConfigurationValidator.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.base.table.options;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Properties;
+
+/**
+ * Interface for classes that validate connector specific table options, including common utils for
+ * validation.
+ */
+@PublicEvolving
+public interface ConfigurationValidator {
+
+    /// Interface
+    Properties getValidatedConfigurations();
+
+    /// Utils
```

Comment:
Did you verify there are no existing alternatives. Surely there must be! I cannot find any though.

Reply:
These utils are moved from the legacy connector module to this more general one, will recheck if any other alternatives could be reused though.